### ISSUE TITLE
Rails/InverseOf: Specify an `:inverse_of` option.

### ISSUE THREAD

**Issue by johnunclesam** [rubocop-hq/rubocop#6059]
## Expected behavior

Doesn't warn at all. I think.

## Actual behavior

It warns me.

## Steps to reproduce the problem

I have this model:

```ruby
class Team < ApplicationRecord
  has_many :players, -> { order 'id DESC' }, dependent: :destroy
end
```

If I remove this: `, -> { order 'id DESC' }` like this:

```ruby
class Team < ApplicationRecord
  has_many :players, dependent: :destroy
end
```

it doesn't warn anymore.

Plus I read that Rails 5 automagically detects `inverse_of` options when no custom options. Right?

So is this a false positive?

## RuboCop version

```
$ rubocop -V
0.57.2
```

**Comment by wata727** [comment]
This isn't a false positive. In 4.2 and later, an inverse association was detected automatically in some cases. But in case of receiving proc, `inverse_of` must be specified. For example, in the following example the same instance is referenced.

```ruby
def Author < ApplicationRecord
  has_many :books
end
```

```
irb(main):001:0> author = Author.first
irb(main):002:0> book = author.books.build
irb(main):003:0> author.equal? book.author
=> true
```

However, in the following example different instances are referenced.

```ruby
def Author < ApplicationRecord
  has_many :books, -> () { order(:created_at) }
end
```

```
irb(main):001:0> author = Author.first
irb(main):002:0> book = author.books.build
irb(main):003:0> author.equal? book.author
  Author Load (0.4ms)  SELECT  "authors".* FROM "authors" WHERE "authors"."id" = ? LIMIT 1  [["id", 1]]
=> false
```

If `inverse_of` is specified, the same instance is referenced.

```ruby
def Author < ApplicationRecord
  has_many :books, -> () { order(:created_at) }, inverse_of: :author
end
```

```
irb(main):001:0> author = Author.first
irb(main):002:0> book = author.books.build
irb(main):003:0> author.equal? book.author
=> true
```

**Comment by johnunclesam** [comment]
Amazing. Thanks.
